File: 2024/problem_list_0010/Java/leetcode/src/test/java/org/renerod1/leetcode/problem_list_10/AppTest.java
package org.renerod1.leetcode.problem_list_10;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class AppTest {

    App app = new App();

    @Test
    public void case1() {
        assertEquals(false, app.isMatch("aa", "a"));
    }

    @Test
    public void case2() {
        assertEquals(true, app.isMatch("aa", "a*"));
    }

    @Test
    public void case3() {
        assertEquals(true, app.isMatch("ab", ".*"));
    }

    @Test
    public void case4() {
        assertEquals(true, app.isMatch("abc", "a***abc"));
    }
}

File: 2024/problem_list_0010/Kotlin/leetcode/src/test/kotlin/org/renerod1/leetcode/problem_list_10/AppTest.kt
package org.renerod1.leetcode.problem_list_10

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class AppTest {

    private lateinit var app: App

    @BeforeEach
    fun configureApp() {
        app = App()
    } 

    @Test
    fun case1() {
        assertEquals(false, app.isMatch("aa", "a"))
    }

    @Test
    fun case2() {
        assertEquals(true, app.isMatch("aa", "a*"))
    }

    @Test
    fun case3() {
        assertEquals(true, app.isMatch("ab", ".*"))
    }

    @Test
    fun case4() {
        assertEquals(true, app.isMatch("abc", "a***abc"))
    }
}